index.js

```diff
@@ -75,7 +75,7 @@ function compression (options) {
 
     // proxy
 
-    res.write = function write (chunk, encoding) {
+    res.write = function write (chunk, encoding, cb) {
       if (ended) {
         return false
```

Comment:
Should the `cb` be called in this case? What does Node.js core do?

Reply:
cutting the circuit is right, something similar [here](https://github.com/nodejs/node/blob/3350d9610864af3219de7dd20e3ac18b3c214c52/lib/_http_outgoing.js#L356)

Reply:
Thanks for that source link! I'm not sure I'm following the connection between res.write and that writeRaw function. Setting up a test server and calling .end then .write with a cb seems to pass an error to the cb.

Reply:
as I understand it should have the same signature as https://github.com/nodejs/node/blob/3350d9610864af3219de7dd20e3ac18b3c214c52/lib/_http_outgoing.js#L761

Reply:
That shows that res.write calls the function `write_` which does indeed call the callback even when finished.